src/AutoRest.CSharp/LowLevel/Output/OperationMethodChainBuilder.cs

```diff
@@ -116,12 +116,15 @@ private bool IsParameterTypeSame(Parameter? first, Parameter? second)
 
         private ReturnTypeChain BuildReturnTypes()
         {
-            var operationBodyTypes = Operation.Responses.Select(r => r.BodyType).WhereNotNull().Distinct().ToArray();
-            CSharpType? responseType = operationBodyTypes.Length switch
+            var operationBodyTypes = Operation.Responses.Where(r => !r.IsErrorResponse).Select(r => r.BodyType).Distinct().ToArray();
+            CSharpType? responseType = null;
+            if (operationBodyTypes != null && operationBodyTypes.Length != 0)
             {
-                0 => null,
-                1 => _typeFactory.CreateType(operationBodyTypes[0]),
-                _ => new CSharpType(typeof(object))
+                var firstBodyType = operationBodyTypes[0];
+                if (firstBodyType != null)
+                {
+                    responseType = _typeFactory.CreateType(firstBodyType);
+                }
```

Comment:
This is changing the previous logic.
In previous implementation, if there are multiple body types, we will use `typeof(object)` as the return type.
Now we will always use the first as return type.
Please double confirm this is expected. At least I believe when there are multiple in the result, and we are taking the first, we should print a warning somewhere

Reply:
It is expected. We cannot handle more than one return types.